modules/core/test/encoder/SegmentEncoder.test.ts

```diff
@@ -34,6 +34,34 @@ describe('encoder->SegmentEncoder', (): void => {
 
   });
 
+  it('should ensure that (en/de)coded (in/out)put restrictions are the same', async (): Promise<void> => {
+
+    const tcModel: TCModel = getTCModelWithGVL();
+    let encoded = '';
+    let decoded: TCModel;
+
+    const encodeIt = (): void => {
+
+      encoded = SegmentEncoder.encode(tcModel, Segment.CORE);
+
+    };
+
+    const decodeIt = (): void => {
+
+      decoded = SegmentEncoder.decode(encoded, tcModel, Segment.CORE);
+
+    };
+
+    await tcModel.gvl.readyPromise;
+
+    expect(encodeIt, 'encode should not throw an error').not.to.throw();
```

Comment:
Hi, can the the unit tests be refactored so that there is only one way it can fail?  This can make potential issues easier to detect.